dali/kernels/erase/erase_gpu.h

```diff
@@ -0,0 +1,526 @@
+    return req;
+  }
+
+  void Run(KernelContext &ctx,
+           OutListGPU<T, ndim> &out,
+           const InListGPU<T, ndim> &in,
+           const InListGPU<ibox<ndim>, 1> &erased_regions,
+           const std::vector<T> &fill_values = {0}) {
+    auto stream = ctx.gpu.stream;  // MAYBE some runtime check if it's not used?
+    const int num_samples = in.num_samples();
+    const int num_fill_values = fill_values.size();
+
+
+    ivec<ndim> region_dim;
+    int erase_regions_count = erased_regions.shape[0][0];
+
+    // Prepare Dim {2, 2, ..., 2, 64, 64}
+    for (int d = 0; d < ndim - 2; d++) {
+      region_dim[d] = 2;
+    }
+    if (ndim > 2) {
+      region_dim[ndim - 2] = 64;
+      region_dim[ndim - 1] = 128;
+      if (channel_dim >= 0) {
+        const int channels = in.shape[0][channel_dim];
+        if (channel_dim == ndim - 1) {
+          region_dim[ndim - 3] = region_dim[ndim - 2];
+          region_dim[ndim - 2] = region_dim[ndim - 1];
+          region_dim[ndim - 1] = channels;
+        } else if (channel_dim == ndim - 2) {
+          region_dim[ndim - 3] = region_dim[ndim - 2];
+          region_dim[ndim - 2] = channels;
+        } else {
+          region_dim[channel_dim] = channels;
+        }
+      }
+    } else if (ndim == 2) {
+      region_dim[0] = 64;
+      region_dim[1] = 128;
+      if (channel_dim >= 0) {
+        const int channels = in.shape[0][channel_dim];
+        region_dim[channels] = channels;
+      }
+    } else {
+      region_dim[0] = 32 * 32;
+    }
+
+
+    int max_regions = 1;
+    for (int i = 0; i < in.num_samples(); i++) {
+      auto region_cover = get_region_cover(region_dim, to_ivec(in.shape[i]));
+      auto total_regions = volume(region_cover);
+      // std::max was complaining
+      max_regions = max_regions >= total_regions ? max_regions : total_regions;
+    }
+    dim3 grid_dim = {(uint32_t)max_regions, (uint32_t)num_samples, 1};
+    dim3 block_dim = {32, 32, 1};  // fixed block dim
+    auto* sample_desc_cpu = ctx.scratchpad->Allocate<sample_t>(AllocType::Host, num_samples);
+    auto* fill_values_cpu = ctx.scratchpad->Allocate<T>(AllocType::Host, num_fill_values);
+    auto* sample_desc_gpu = ctx.scratchpad->Allocate<sample_t>(AllocType::GPU, num_samples);
+    auto* fill_values_gpu = ctx.scratchpad->Allocate<T>(AllocType::GPU, num_fill_values);
+
+    for (int i = 0; i < num_samples; i++) {
+      sample_desc_cpu[i].in = in.data[i];
+      sample_desc_cpu[i].out = out.data[i];
+      sample_desc_cpu[i].erase_regions = erased_regions.data[i];
+      for (int dim = 0; dim < ndim; dim++) {
+        sample_desc_cpu[i].sample_shape[dim] = in.tensor_shape_span(i)[dim];
+      }
+      sample_desc_cpu[i].sample_stride = GetStrides(sample_desc_cpu[i].sample_shape);
+    }
+    memcpy(fill_values_cpu, fill_values.data(), num_fill_values * sizeof(T));
+
+    CUDA_CALL(
+      cudaMemcpyAsync(sample_desc_gpu, sample_desc_cpu, num_samples * sizeof(sample_t),
```

Comment:
you can get away with a single `cudaMemcpyAsync`, since the memory from the scratchpad is contiguous

Reply:
Scratchpad::ToGPU to the rescue!

Reply:
Didn't we have a magic one that was supposed to do that in one go?
Still I think the cudaMemcpy is a bit clearer than ToGPU that's a bit cryptic and doesn't have any docstring. 
Similar thing to the div_ceil vs get_region_cover above

Reply:
I 👍 for ToGPU, if we have it we should use it and be consistent. Or remove `ToGPU` at all.